assets/js/user_form/index.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * A hook for the user profile form.
+ *
+ * On submit this hook saves the new data into cookie.
+ * This cookie serves as a backup and can be used to restore
+ * user data if the server is restarted.
+ */
+const UserForm = {
+  mounted() {
+    this.el.addEventListener("submit", (event) => {
+      const name = this.el.data_name.value;
+      const color = this.el.data_color.value;
+      storeUserData({ name, color });
+    });
+  },
+};
+
+function storeUserData(userData) {
+  const value = JSON.stringify(userData);
+  setCookie("user_data", value, 157680000); // 5 years
+}
+
+function setCookie(key, value, maxAge) {
+  const cookie = `${key}=${value};max-age=${maxAge};path=/`;
```

Comment:
We may need to escape the value. What if the value has a `=` or `;` in it? Perhaps we need to encode it with base64 or base16? Is there an easy way to do that with JS?

Reply:
Good catch! There is `btoa` that does base64 encoding, but it handles only latin1 characters. Fortunately there's a trick for that ^^

```js
function encodeBase64(string) {
  return btoa(unescape(encodeURIComponent(string)));
}
```